pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -292,6 +292,19 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1beta1.PipelineRun) err
 		return nil
 	}
 
+	// build DAG with a list of final tasks, this DAG is used later to identify
+	// if a task in PipelineRunState is final task or not
+	// the finally section is optional and might not exist
```

Comment:
i just realized we don't have a section for finally tasks in the pipelinerun status and we have not changed the status at all.

@pritidesai would it make the code simpler if you could treat the statuses of the dag tags and finally tasks separately? If so, i think it would be worth proposing a new section in status and including that in this PR.

if you don't think it'll be simpler (and if it will actually be more complicated) then nvm

Reply:
pipelinerun status stays the same, code wouldnt simplify even if we add such separation.

Also, it wouldnt make sense to treat them separately as they (DAG and finally) are part of one single pipeline. When DAG tasks are done executing, pipelinerun status is set to `Success` but this same status would change to `Failure` if a finally task fails. This will create confusion when people see events getting generated with `Running` -> `Success` -> `Failed` i.e. how can a pipeline fail after finishing with success?

Also, from a user perspective, I have one single pipeline, and would expect Done (Success/Failure) once all the tasks are finished including (DAG and finally). 

This kind of separation would create even more confusion when we add pipeline level `exception`.

Sorry @bobcatfish but I would not change the status based on the sections (`tasks` and `finally`). I dont think it will be simpler 🙏

Reply:
In this PR we kept one `pipelineState` which now is mapped to two DAGs. 
This is the reason that several `pipelineState` functions now have to take a DAG as an input.

Some of these functions have slightly different implementations depending on which DAG is passed in:

```
isDAGTasksDone(dag) # only works if the main DAG is passed in
isFinalTasksDone(dag) # only works if the final DAG is passed in
```

which is one of the reasons in https://github.com/tektoncd/pipeline/pull/2821 I decided to associate the dag in the state in `pipelineState`. Once this is merged, both DAGs will be in the `pipelineState` I guess. That will remove the need to pass the DAG into this functions. It works from a code POV, I'm not sure it's the best or more logical solution, I'm open to suggestions.

I'd really like to end up having code in this module where the function names suggest what they do, hopefully https://github.com/tektoncd/pipeline/pull/2821 will help with that.